presto-main/src/main/java/io/prestosql/event/QueryMonitor.java

```diff
@@ -309,15 +325,34 @@ private QueryContext createQueryContext(SessionRepresentation session, Optional<
 
     private static QueryIOMetadata getQueryIOMetadata(QueryInfo queryInfo)
     {
+        Multimap<FragmentNode, OperatorStats> tableOperatorStats = extractStatsForTableScan(queryInfo);
+
         ImmutableList.Builder<QueryInputMetadata> inputs = ImmutableList.builder();
         for (Input input : queryInfo.getInputs()) {
+            // Note: input table may be scanned multiple times
+            Collection<OperatorStats> inputTableOperatorStats = tableOperatorStats.get(new FragmentNode(input.getFragmentId(), input.getPlanNodeId()));
```

Comment:
same comment about the comment here: the reason seems to be a plan node corresponding to multiple operators